src/analysis/retail/monk/mistweaver/modules/spells/DancingMists.tsx

```diff
@@ -0,0 +1,235 @@
+  get dancingMistReMHealing() {
+    return this.dancingMistHealing + this.dancingMistAbsorbed;
+  }
+
+  get dancingMistVivifyHealing() {
+    return this.extraVivHealing + this.extraVivAbsorbed;
+  }
+
+  get mistyPeaksHealingFromDancingMist() {
+    return this.extraMistyPeaksHealing + this.extraMistyPeaksAbsorb;
+  }
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.DANCING_MISTS_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.RENEWING_MIST_HEAL),
+      this.onApplyRem,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.RENEWING_MIST_HEAL),
+      this.onReMHeal,
+    );
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.VIVIFY), this.handleVivify);
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.VIVIFY),
+      this.handleVivifyCast,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(TALENTS_MONK.ENVELOPING_MIST_TALENT),
+      this.handleEnvApply,
+    );
+    this.addEventListener(
+      Events.refreshbuff.by(SELECTED_PLAYER).spell(TALENTS_MONK.ENVELOPING_MIST_TALENT),
+      this.handleEnvApply,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(TALENTS_MONK.ENVELOPING_MIST_TALENT),
+      this.onEnvHeal,
+    );
+  }
+
+  onApplyRem(event: ApplyBuffEvent) {
+    const playerId = event.targetID;
+    if (
+      !this.hotTracker.hots[playerId] ||
+      !this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id];
+    if (this.hotTracker.fromDancingMists(hot) && !this.hotTracker.fromBounce(hot)) {
+      this.dancingMistCount += 1;
+    }
+  }
+
+  onReMHeal(event: HealEvent) {
+    const playerId = event.targetID;
+    if (
+      !this.hotTracker.hots[playerId] ||
+      !this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id];
+    if (this.hotTracker.fromDancingMists(hot)) {
+      this.dancingMistHealing += event.amount || 0;
+      this.dancingMistAbsorbed += event.absorbed || 0;
+      this.dancingMistOverhealing += event.overheal || 0;
+    }
+  }
+  handleVivifyCast(event: CastEvent) {
+    this.lastVivifyCastTarget = event.targetID || 0;
+    this.countedMainVivifyHit = false;
+  }
+
+  handleVivify(event: HealEvent) {
+    const targetId = event.targetID;
+    if (
+      !this.hotTracker.hots[targetId] ||
+      !this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    // only count cleave hit on main target
+    if (targetId === this.lastVivifyCastTarget && !this.countedMainVivifyHit) {
+      this.countedMainVivifyHit = true;
+      return;
```

Comment:
```suggestion
    if (this.vivify.lastCastTarget === targetId && this.vivify.mainTargetHitsToCount > 0) {
      return;
```
import vivify module and use its variables that track whether it counted main hit or not